src/config.erl

```diff
@@ -177,9 +182,19 @@ delete(Section, Key, Reason) ->
 
 delete(Sec, Key, Persist, Reason) when is_binary(Sec) and is_binary(Key) ->
     delete(binary_to_list(Sec), binary_to_list(Key), Persist, Reason);
-delete(Section, Key, Persist, Reason) when is_list(Section), is_list(Key) ->
+delete(Section, Key, Persist, Reason) when is_boolean(Persist) ->
+    assert_string(Section),
+    assert_string(Key),
+    if Reason == nil -> ok; true -> assert_string(Reason) end,
     gen_server:call(?MODULE, {delete, Section, Key, Persist, Reason}).
 
+assert_string(Term) ->
+    case io_lib:printable_unicode_list(Term) of
```

Comment:
Hm...I feel sorry for unicode suggestion, but confusion as well. I put console log call before the case in order to print the terms that comes here and here is what I saw:
```
>>> "медвед"
>>> "водка"
>>> "медвед"
>>> "водка"
>>> "балалайка"
2015-06-09 21:01:11.116 [debug] node1@127.0.0.1 <0.1975.0> 'PUT' /_config/медвед/водка {1,1} from "127.0.0.1"
Headers: [{'Accept',"*/*"},{'Content-Length',"20"},{'Content-Type',"application/x-www-form-urlencoded"},{'Host',"localhost:15986"},{'User-Agent',"curl/7.42.1"}]
2015-06-09 21:01:11.117 [debug] node1@127.0.0.1 <0.1975.0> OAuth Params: []
2015-06-09 21:01:11.118 [notice] node1@127.0.0.1 <0.53.0> config: [медвед] водка set to балалайка for reason nil
2015-06-09 21:01:11.121 [notice] node1@127.0.0.1 <0.1975.0> 127.0.0.1 - - PUT /_config/медвед/водка 200
>>> "httpd"
>>> "enable_cors"
>>> "fabric"
>>> "changes_duration"
>>> "rexi"
>>> "server_per_node"
>>> "rexi"
>>> "server_per_node"
>>> "rexi"
>>> "server_per_node"
>>> "httpd"
>>> "x_forwarded_host"
2015-06-09 21:01:21.176 [debug] node1@127.0.0.1 <0.229.0> 'PUT' /_config/тест/тест {1,1} from "127.0.0.1"
Headers: [{'Accept',"*/*"},{'Content-Length',"10"},{'Content-Type',"application/x-www-form-urlencoded"},{'Host',"localhost:15986"},{'User-Agent',"curl/7.42.1"}]
>>> "httpd"
>>> "max_uri_length"
2015-06-09 21:01:21.176 [debug] node1@127.0.0.1 <0.229.0> OAuth Params: []
>>> "couch_httpd_auth"
>>> "require_valid_user"
>>> "httpd"
>>> "config_whitelist"
>>> "couchdb"
>>> "max_document_size"
>>> [209,130,208,181,209,129,209,130]
2015-06-09 21:01:21.177 [error] node1@127.0.0.1 <0.229.0> Badarg error in HTTP request
2015-06-09 21:01:21.177 [info] node1@127.0.0.1 <0.229.0> Stacktrace: [{config,assert_string,1,[{file,"src/config.erl"},{line,197}]},{config,get,3,[{file,"src/config.erl"},{line,141}]},{couch_httpd_misc_handlers,handle_approved_config_req,3,[{file,"src/couch_httpd_misc_handlers.erl"},{line,283}]},{couch_httpd,handle_request_int,5,[{file,"src/couch_httpd.erl"},{line,301}]},{mochiweb_http,headers,5,[{file,"src/mochiweb_http.erl"},{line,93}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,237}]}]
>>> "httpd"
2015-06-09 21:01:21.177 [notice] node1@127.0.0.1 <0.229.0> 127.0.0.1 - - PUT /_config/тест/тест 500
>>> "enable_cors"
2015-06-09 21:01:21.177 [error] node1@127.0.0.1 <0.229.0> httpd 500 error response:
 {"error":"unknown_error","reason":"badarg"}
```

So it seems that `Term` could be both a Unicode string and a list of UTF-8 codepoints depending if some specifics character in string are present. I'm not sure now what causes such behaviour, but it makes assertion fail.